recipes/imagemgagick/all/conanfile.py

```diff
@@ -0,0 +1,349 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from conans import ConanFile, tools, AutoToolsBuildEnvironment, MSBuild
+import os
+import shutil
+import glob
+
+
+class ImageMagicConan(ConanFile):
+    name = "imagemagick"
+    description = ("ImageMagick is a free and open-source software suite for displaying, converting, and editing "
+                   "raster image and vector image files")
+    topics = ("conan", "imagemagick", "images", "manipulating")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/ImageMagick/ImageMagick"
+    license = "ImageMagick"
+    exports = ["LICENSE.md"]
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False],
+               "fPIC": [True, False],
+               "hdri": [True, False],
+               "quantum_depth": [8, 16, 32],
+               "zlib": [True, False],
+               "bzlib": [True, False],
+               "lzma": [True, False],
+               "lcms": [True, False],
+               "openexr": [True, False],
+               "jpeg": [True, False],
+               "openjp2": [True, False],
+               "png": [True, False],
+               "tiff": [True, False],
+               "webp": [True, False],
+               "xml": [True, False],
+               "freetype": [True, False],
+               "utilities": [True, False]}
+    default_options = {"shared": False,
+                       "fPIC": True,
+                       "hdri": True,
+                       "quantum_depth": 16,
+                       "zlib": True,
+                       "bzlib": True,
+                       "lzma": False,
+                       "lcms": False,
+                       "openexr": False,
+                       "jpeg": True,
+                       "openjp2": False,
+                       "png": True,
+                       "tiff": True,
+                       "webp": False,
+                       "xml": False,
+                       "freetype": True,
+                       "utilities": True}
+
+    _source_subfolder = "ImageMagick"  # name is important, VisualMagick uses relative paths to it
+    _build_subfolder = "build_subfolder"
+
+    @property
+    def _is_mingw_windows(self):
+        return self.settings.os == 'Windows' and self.settings.compiler == 'gcc' and os.name == 'nt'
```

Comment:
You should use `tools.os_info.is_windows` here instead of `os.name`.

To support two profile build approach (host and build profile), you might want to use `settings_build` as well.